### DISCUSSION TITLE
Few questions: API limits, migrating users, webhooks

### DISCUSSION THREAD

**Discussion by osseonews** [pocketbase/pocketbase#529] [Q&A]
Thanks again for this great software. A few quesetions:

1. How many API requests per second can you get from Pocketbase? Would the max API requests be a function of the server pocketbase is running on or is there a particular limitation with sqlite? I've read that sqlite is very slow for writes. I only ask this question b/c I wonder how scalable Pocketbase would be as a backend for nextjs website, both for static generation (nextjs can make a ton of concurrent requests to a data source when building out the static pages) and client-side requests to update data.

2. How are passwords stored in Pocketbase? If we were to migrate our users to Pocketbase, and their passwords are hashed with bcrypt, would the migration work or would all the users have to reset their passwords? 

3. Does pocketbase have webhooks? I can't seem to find any documentation on them, though I did notice you have Event hooks, but presumably you need to know how to program in Go to create a webhook based on event hooks?

**Comment by ganigeorgiev** [comment]
1. There is no official benchmark yet. Once a more stable version is released, I'll try to publish a testbed and some benchmarks. I've only local tests with Hetzner CX21 instance and I was able to serve more than 10000+ concurrent/simultaneous realtime connections (only ulimit need to be adjusted). SQLite supports only a single writer at a time (multiple writes queue up) but are quite fast and as long as you don't have a long lasting write transaction (which we don't) it should be ok for most use cases.

2. PocketBase uses bcrypt for the users and admins passwords. Migrating users from other platforms that uses bcrypt doesn't require resetting the users password.

3. Currently only if you use PocketBase as framework (see [event hooks](https://pocketbase.io/docs/event-hooks/)). For a future release, there are plans for collection triggers - #317.
